libraries/AC_Fence/AC_Fence.cpp

```diff
@@ -254,6 +313,54 @@ bool AC_Fence::check_fence_alt_max()
     return false;
 }
 
+/// returns true if we have freshly breached the minimum altitude
+/// fence; also may set up a fallback fence which, if breached, will
+/// cause the altitude fence to be freshly breached
+bool AC_Fence::check_fence_alt_min()
+{
+    // altitude fence check
+    if (!(_enabled_fences & AC_FENCE_TYPE_ALT_MIN)) {
+        // not enabled
+        return false;
+    }
+
+    AP::ahrs().get_relative_position_D_home(_curr_alt);
+    _curr_alt = -_curr_alt; // translate Down to Up
+
+    // check if we are under the altitude fence
+    if(_curr_alt < _alt_min) {
+
+        // record distance below breach
+        _alt_min_breach_distance = _alt_min - _curr_alt;
+
+        // check for a new breach or a breach of the backup fence
+        if (!(_breached_fences & AC_FENCE_TYPE_ALT_MIN) ||
+            (!is_zero(_alt_min_backup) && _curr_alt < _alt_min_backup)) {
+            
+            // new breach
+            record_breach(AC_FENCE_TYPE_ALT_MIN);
+
+            // create a backup fence 20m lower down
+            _alt_min_backup = _curr_alt - AC_FENCE_ALT_MIN_BACKUP_DISTANCE;
+            // new breach
+            return true;
+        }
+        // old breach
+        return false;
+    }
+
+    // not breached
+
+    // clear alt breach if present
+    if ((_breached_fences & AC_FENCE_TYPE_ALT_MIN) != 0) {
+        clear_breach(AC_FENCE_TYPE_ALT_MIN);
+        _alt_min_backup = 0.0f;
```

Comment:
Do we need to zero these?

Reply:
Implementation for `check_fence_alt_min` is a reworked `check_fence_alt_max`, which contains this same assignment. I'd have to confirm the conditions in which they backup fence is used.

I suspect if we dont, the following series may occur:

- Flying within fences.
- Breach floor. Set backup floor
- Breach backup floor. Set new backup floor (repeat `n` times)
- Return inside fence
- Breach `FENCE_ALT_MIN`, `(!is_zero(_alt_min_backup) && _curr_alt < _alt_min_backup)` is treated as and old breach, because our current alt is less than `ALT_MIN` but greater than the `alt_min_backup`.